test/SILOptimizer/existential_transform_extras_ossa.sil

```diff
@@ -261,3 +281,31 @@ bb0(%0 : @guaranteed $PlotView):
   %v = tuple ()
   return %v : $()
 }
+
+// CHECK-LABEL: sil shared [ossa] @$s35testExistentialSpecializeGuaranteedTf4e_n : 
+// CHECK:  [[VAL1:%.*]] = init_existential_ref %0
+// CHECK:  [[VAL2:%.*]] = open_existential_ref [[VAL1]]
+// CHECK-LABEL: } // end sil function '$s35testExistentialSpecializeGuaranteedTf4e_n'
+sil hidden [ossa] @testExistentialSpecializeGuaranteed : $@convention(thin) (@guaranteed Klass3 & P) -> Int32 {
+bb0(%0 : @guaranteed $Klass3 & P):
+  %2 = open_existential_ref %0 : $Klass3 & P to $@opened("77949BFA-77BC-11EB-BC0E-F2189810406F") Klass3 & P
+  %3 = alloc_stack $@opened("77949BFA-77BC-11EB-BC0E-F2189810406F") Klass3 & P
+  %4 = store_borrow %2 to %3 : $*@opened("77949BFA-77BC-11EB-BC0E-F2189810406F") Klass3 & P
```

Comment:
Can you add more tests here? Also I think that we should be handle more complex guaranteed cases such as:

```
diff --git a/test/SILOptimizer/existential_transform_extras_ossa.sil b/test/SILOptimizer/existential_transform_extras_ossa.sil
index f2e54d7b650..2b1724ddfde 100644
--- a/test/SILOptimizer/existential_transform_extras_ossa.sil
+++ b/test/SILOptimizer/existential_transform_extras_ossa.sil
@@ -300,12 +300,37 @@ bb0(%0 : @guaranteed $Klass3 & P):
   return %6 : $Int32
 }
 
+// CHECK-LABEL: sil shared [ossa] @$s36testExistentialSpecializeGuaranteed2Tf4e_n : 
+// CHECK:  [[VAL1:%.*]] = init_existential_ref %0
+// CHECK:  [[VAL2:%.*]] = open_existential_ref [[VAL1]]
+// CHECK-LABEL: } // end sil function '$s36testExistentialSpecializeGuaranteed2Tf4e_n'
+sil hidden [ossa] @testExistentialSpecializeGuaranteed2 : $@convention(thin) (@owned Klass3 & P) -> Int32 {
+bb0(%0 : @owned $Klass3 & P):
+  %0a = begin_borrow %0 : $Klass3 & P
+  %2 = open_existential_ref %0a : $Klass3 & P to $@opened("77949BFA-77BC-11EB-BC0E-F2189810406F") Klass3 & P
+  %3 = alloc_stack $@opened("77949BFA-77BC-11EB-BC0E-F2189810406F") Klass3 & P
+  %4 = store_borrow %2 to %3 : $*@opened("77949BFA-77BC-11EB-BC0E-F2189810406F") Klass3 & P
+  %5 = witness_method $@opened("77949BFA-77BC-11EB-BC0E-F2189810406F") Klass3 & P, #P.foo : <Self where Self : P> (Self) -> () -> Int32, %2 : $@opened("77949BFA-77BC-11EB-BC0E-F2189810406F") Klass3 & P : $@convention(witness_method: P) <τ_0_0 where τ_0_0 : P> (@in_guaranteed τ_0_0) -> Int32
+  %6 = apply %5<@opened("77949BFA-77BC-11EB-BC0E-F2189810406F") Klass3 & P>(%3) : $@convention(witness_method: P) <τ_0_0 where τ_0_0 : P> (@in_guaranteed τ_0_0) -> Int32
+  dealloc_stack %3 : $*@opened("77949BFA-77BC-11EB-BC0E-F2189810406F") Klass3 & P
+  end_borrow %0a : $Klass3 & P
+  destroy_value %0 : $Klass3 & P
+  return %6 : $Int32
+}
+
+sil @mixInt : $@convention(thin) (Int32, Int32) -> Int32
+
 sil hidden [ossa] @entrypoint : $@convention(thin) (@guaranteed Klass4) -> Int32 {
 bb0(%0 : @guaranteed $Klass4):
   %1 = init_existential_ref %0 : $Klass4 : $Klass4, $Klass3 & P
   %func = function_ref @testExistentialSpecializeGuaranteed : $@convention(thin) (@guaranteed Klass3 & P) -> Int32
   %res = apply %func(%1) : $@convention(thin) (@guaranteed Klass3 & P) -> Int32
-  return %res : $Int32
+  %func2 = function_ref @testExistentialSpecializeGuaranteed2 : $@convention(thin) (@owned Klass3 & P) -> Int32
+  %1a = copy_value %1 : $Klass3 & P
+  %res2 = apply %func2(%1a) : $@convention(thin) (@owned Klass3 & P) -> Int32
+  %mixFunc = function_ref @mixInt : $@convention(thin) (Int32, Int32) -> Int32
+  %res3 = apply %mixFunc(%res, %res2) : $@convention(thin) (Int32, Int32) -> Int32
+  return %res3 : $Int32
 }
 
 sil_witness_table hidden Klass4: P module devirtualize_protocol_composition {
```

Can you also check that we can handle from a load_borrow and as above begin_borrow?